### ISSUE TITLE
Python 3.8 instead of Python 3.7

### ISSUE THREAD

**Issue by eirnym** [macvim-dev/macvim#1012]
It's time to move on to python 3.8. It was released half a year ago and vim core supports it. Most libraries which had supported python 3.7 also have support for python 3.8

**Comment by ychin** [comment]
Homebrew is still at 3.7.6: https://formulae.brew.sh/formula/python

There are some discussions here (https://discourse.brew.sh/t/why-python-3-8-has-not-been-updated-yet/6057/16) and there is a ticket (https://github.com/Homebrew/homebrew-core/issues/47274) tracking this migration.

Given that Homebrew hasn't switched over I prefer to not upgrade until they do it. Part of the reason is that it complicates our CI build scripts and also I think we should assume most users install the default `python` package instead of the explicit `python@3.8`.

**Comment by eirnym** [comment]
I still don't agree with an idea "all ports should use one _true_ python/java/(any other tool)". I'm asking about only Github version. 

My personal opinion, that if Homebrew wants to stay behind using following ideology of "one true ..", they can patch `vimrc` and other defaults as they pleased.

**Comment by eirnym** [comment]
I understand, that my proposition is not very nice, but Homebrew as any other package manager has much greater mobility to change params, code, and other things than users who use releases from Github like me. 

Unfortunately, simply specifying python3.8 in a library path doesn't work well for me.

**Comment by eirnym** [comment]
@ychin BTW, Your response sounds like "you use Homebrew/MacPorts to use MacVim with Python3.8 (and more likely 3.7) or build and maintain it manually".

**Comment by ychin** [comment]
So can you be more specific what supporting "Python 3.8" means?

Are you talking about how we automatically set `pythonthreedll` to be 3.7? Can't you just set that variable to the correct path yourself. The auto-detection is really a clutch because there's no way for Vim to be able to find all possible Python versions for you.

Or did you mean something else?

**Comment by eirnym** [comment]
I can't just change `pythonthreedll` as ABI is not compatible between python versions(third number is compatible while second may be not). [PEP-606](https://www.python.org/dev/peps/pep-0606/) to provide backward binary compatibility was rejected.

So I need MacVim to be compiled with python3.8 and I'd like to update official `vimrc` from `3.7` to `3.8`, that's it.

As I've said, Homebrew, Macports and any other port system may stay with compatibility ranges of MacVim by using little patch to `vimrc` and by provide an appropriate argument to configure script. Alternatively to make manual patches, configure/make script itself may patch `vimrc`,

**Comment by eirnym** [comment]
They already changed the python to 3.8 across the Homebrew.

**Comment by eirnym** [comment]
Nevertheless, I ask you for making official builds with 2-3 major python releases when it's possible. Other languages doesn't change so much as Python nowadays. This will give an opportunity to use an official build and use Python version they want. 

For Homebrew only one build will not require an additional dependency installation, so price is small.

**Comment by radome** [comment]
Homebrew migration to Python 3.8 is complete: https://github.com/Homebrew/homebrew-core/issues/47274#issuecomment-656361893
Latest MacVim snapshot [163](https://github.com/macvim-dev/macvim/releases/tag/snapshot-163) which is installed by hombrew-cask seems to be pointing to Python 3.7

**Comment by itsayellow** [comment]
`:python3` in my homebrew cask installed macvim is no longer operational because it doesn't find python3.7

```
E370: Could not load library /usr/local/Frameworks/Python.framework/Versions/3.7/Python
E263: Sorry, this command is disabled, the Python library could not be loaded.
```

**Comment by itsayellow** [comment]
Maybe in the future MacVim can default to the 'Current' subdirectory instead of a hard-coded version-numbered subdirectory, like:
```
set pythonthreedll=/usr/local/Frameworks/Python.framework/Versions/Current/Python
```

**Comment by eirnym** [comment]
> When using homebrew's python, maybe in the future MacVim can default to the 'Current' subdirectory instead of a hard-coded version-numbered subdirectory, like:
> 
> ```
> set pythonthreedll=/usr/local/Frameworks/Python.framework/Versions/Current/Python
> ```

No, it's impossible to do as I wrote above Python resigned from PEP-606 to support backward binary compatibility. So you'll have problems with Python 3.7 used in MacVim & Python 3.8/3.9 etc.

**Comment by eirnym** [comment]
> `:python3` in my homebrew cask installed macvim is no longer operational because it doesn't find python3.7
> 
> ```
> E370: Could not load library /usr/local/Frameworks/Python.framework/Versions/3.7/Python
> E263: Sorry, this command is disabled, the Python library could not be loaded.
> ```

this is one of problems I want to avoid

**Comment by martensson** [comment]
I am getting the same error now and unable to use Macvim after homebrew upgraded to 3.8.

**Comment by ychin** [comment]
I'll fix this in the weekend. But yes, the current way we are handling it is non-ideal.

**Comment by radome** [comment]
> I am getting the same error now and unable to use Macvim after homebrew upgraded to 3.8.

I've resolved with: 
```
brew install python@3.7
``` 
then 
```
ln -s /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7 /usr/local/Frameworks/Python.framework/Versions/3.7
```

**Comment by astyagun** [comment]
Another possible workaround:

```vim
set pythonthreedll=/usr/local/Frameworks/Python.framework/Versions/Current/Python
```

**Comment by itsayellow** [comment]
> Another possible workaround:
> 
> ```viml
> set pythonthreedll=/usr/local/Frameworks/Python.framework/Versions/Current/Python
> ```

I was using that for a while but started getting weird errors, it may have been some sort of binary incompatibility as mentioned by @eirnym above.

**Comment by eirnym** [comment]
> > I am getting the same error now and unable to use Macvim after homebrew upgraded to 3.8.
> 
> 
> 
> I've resolved with: 
> 
> ```
> 
> brew install python@3.7
> 
> ``` 
> 
> then 
> 
> ```
> 
> ln -s /usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7 /usr/local/Frameworks/Python.framework/Versions/3.7
> 
> ```

It's also possible to set pythonthreedll to `/usr/local/Cellar/python@3.7/3.7.8_1/Frameworks/Python.framework/Versions/3.7` in your .vimrc

**Comment by ychin** [comment]
I believe @ichizok fixed this in d552421a21 already, and #1063 will make this compatible with binary install and MacPort as well. I think the issue here is the we haven't made a release yet so if you are using Homebrew Cask / official MacVim binary release you wouldn't have obtained that update that is built against 3.8 yet. I will try to make an update soon to address this.

**Comment by ychin** [comment]
Closing as version 164 is out.
